editoast/src/views/stdcm_logs.rs

```diff
@@ -0,0 +1,206 @@
+use axum::extract::Path;
+use axum::extract::State;
+use axum::Extension;
+use axum::Json;
+use editoast_authz::BuiltinRole;
+use editoast_derive::EditoastError;
+use editoast_models::DbConnectionPoolV2;
+use thiserror::Error;
+use utoipa::IntoParams;
+
+use crate::error::Result;
+use crate::models::stdcm_log::StdcmLog;
+use crate::List;
+use crate::SelectionSettings;
+
+use super::AuthenticationExt;
+use super::AuthorizationError;
+
+crate::routes! {
+    "/stdcm_logs" => {
+        list_stdcm_logs,
+        "/{trace_id}" => {
+            stdcm_log_by_trace_id,
+        },
+    },
+}
+
+#[derive(Debug, Error, EditoastError)]
+#[editoast_error(base_id = "stdcm_log")]
+pub enum StdcmLogError {
+    #[error("StdcmLog '{trace_id}', could not be found")]
+    #[editoast_error(status = 404)]
+    NotFound { trace_id: String },
+}
+
+#[utoipa::path(
+    get, path = "",
+    tag = "stdcm_log",
+    responses(
+        (status = 200, body = Vec<StdcmLog>, description = "The list of Stdcm Logs"),
+    )
+)]
+async fn list_stdcm_logs(
+    State(db_pool): State<DbConnectionPoolV2>,
+    Extension(auth): AuthenticationExt,
+) -> Result<Json<Vec<StdcmLog>>> {
+    let authorized = auth
+        .check_roles([BuiltinRole::Superuser].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Forbidden.into());
+    }
+
+    let conn = &mut db_pool.get().await?;
+
+    let stdcm_logs = StdcmLog::list(conn, SelectionSettings::new()).await?;
+
+    Ok(Json(stdcm_logs))
+}
+
+#[derive(IntoParams)]
+#[allow(unused)]
+pub struct StdcmLogTraceIdParam {
+    trace_id: String,
```

Comment:
There's no notion of `trace` in the API.

```suggestion
    id: String,
```

Reply:
Maybe the struct can be renamed too?

Reply:
Maybe I misunderstood the comment, but the purpose of the endpoint is to search by `trace_id`, not by the stdcm_log `id`.

Reply:
Mmmh, I believe you're right indeed. So maybe it makes sense to expose the notion of "trace" in the API. If I recall correctly, the idea is to detect a problem through our observability stack (OpenTelemetry), and therefore find a trace ID, then ask `editoast` for more information about it (STDCM logs). Do you agree @leovalais? We might be able to resolve that comment and leave the code as is?

Reply:
You're right. These are just IDs when looking just at the API, but since this is the goal workflow let's keep `trace_id`.